app/controllers/comments_controller.rb

```diff
@@ -322,6 +322,12 @@ def edit
   # POST /comments
   # POST /comments.xml
   def create
+    # If the commentable is a story or chapter, and it is not posted, then redirect with flash error
+    if @commentable.respond_to?(:posted?) && !@commentable.posted?
+      flash[:error] = ts("You cannot comment on an unposted work.")
+      redirect_back_or_to root_path and return
+    end
```

Comment:
Instead of just blocking creation, it would be better to block all comment changes on draft works, similar to hidden and unrevealed works:

https://github.com/otwcode/otwarchive/blob/f712d25a41998fdc17e9e0c261268908043d6521/app/controllers/comments_controller.rb#L72

The message should also be translatable.